src/components/Analytics/utils.ts

```diff
@@ -97,7 +97,7 @@ function toTitleCase(name: string) {
 
 // Capitalize string (special characters kept in place by not using "words" from lodash)
 function toTitleCaseWithSpecialCharacters(name: string) {
-  return name.split(' ').map(capitalize).join(' ');
+  return name ? name.split(' ').map(capitalize).join(' ') : '';
```

Comment:
You said:
> our new toTitleCaseWithSpecialCharacters function was not allowing undefined as an argument although it should have

But `name` is defined as `string`, so:

1. We should change the type to `name | undefined` I guess.
2. We would ideally figure out why TypeScript was letting us pass `undefined` to a function that declares its type as `string`.  Normally that would be an error and we couldn't have shipped this code.  So we may be doing something "wrong" somewhere upstream of this function too.